go/ocr2/decryptionplugin/decryption.go

```diff
@@ -82,6 +82,7 @@ func (dp *decryptionPlugin) Query(ctx context.Context, ts types.ReportTimestamp)
 	for _, request := range decryptionRequests {
 		ciphertext := &tdh2easy.Ciphertext{}
 		if err := ciphertext.UnmarshalVerify(request.Ciphertext, dp.publicKey); err != nil {
+			dp.decryptionQueue.SetResult(request.CiphertextId, nil, fmt.Errorf("cannot unmarshal the ciphertext: %w", err))
```

Comment:
Since the errors are user-facing, I think we should avoid returning the full error message.  Instead, let's use something similar to the following error message:

```suggestion
			dp.decryptionQueue.SetResult(request.CiphertextId, nil, errors.New("cannot decrypt secrets: encrypted bytes are invalid"))
```

Reply:
@KuphJr user-facing meaning? We don't have to pass them to the end user, we can always send a different error with a response after processing this one.

@stchrysa  can we export them as named errors (like this: https://github.com/smartcontractkit/libocr/blob/master/ragep2p/internal/knock/knock.go#L40)

Reply:
@KuphJr I prefer we do not miss the information of the inner error, in case this interface is used by some other application in the future.

Reply:
@bolekk with named errors we can also not pass the inner error, right?
Given that we log it it might be ok.
Wdyt?

Reply:
I thought again about and named errors seem ok, since we log the wrapped error anyway.
See the latest commit.